client/me/purchases/manage-purchase/index.tsx

```diff
@@ -639,6 +642,76 @@ class ManagePurchase extends Component<
 		return null;
 	}
 
+	renderRefundText() {
+		const { purchase, translate } = this.props;
+
+		// Hide if refund window has lapsed.
+		if ( ! purchase?.isRefundable || ! purchase?.refundText || ! purchase?.mostRecentRenewDate ) {
+			return;
+		}
+
+		const ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+		const lastRenewalDate = new Date( purchase.mostRecentRenewDate );
+		const refundPeriodEndDate = new Date(
+			lastRenewalDate.getTime() + purchase.refundPeriodInDays * ONE_DAY_IN_MILLISECONDS
+		);
+
+		const refundText = (() => {
+			// User would receive another free domain credit upon cancelling.
+			if ( isDomainRegistration( purchase ) && ! hasAmountAvailableToRefund( purchase ) ) {
```

Comment:
I'm not sure this is an accurate way to figure out if a domain credit was used.  I think it could also pick up other cases where a domain was free (or even paid but then already refunded).

Reply:
Ah, that makes sense. Would you have any suggestions about the best way to check this? Although I suppose that wouldn’t be necessary if domain credits aren’t returned (see below!).